database/engine/rrdengineapi.c

```diff
@@ -917,15 +921,19 @@ int rrdeng_init(RRDHOST *host, struct rrdengine_instance **ctxp, char *dbfiles_p
 
         rrd_stat_atomic_add(&global_fs_errors, 1);
         rrd_stat_atomic_add(&rrdeng_reserved_file_descriptors, -RRDENG_FD_BUDGET_PER_INSTANCE);
-        return UV_EMFILE;
+        return NULL;//UV_EMFILE;
     }
+    char dbfiles_path[FILENAME_MAX + 1];
+    int ret;
 
-    if (NULL == ctxp) {
-        ctx = &multidb_ctx;
-        memset(ctx, 0, sizeof(*ctx));
-    } else {
-        *ctxp = ctx = callocz(1, sizeof(*ctx));
-    }
+    snprintfz(dbfiles_path, FILENAME_MAX, "%s/dbengine", host->cache_dir);
+    ret = mkdir(dbfiles_path, 0775);
```

Comment:
database/engine/rrdengineapi.c: In function 'rrdeng_init':
database/engine/rrdengineapi.c:927:9: warning: variable 'ret' set but not used [-Wunused-but-set-variable]
  927 |     int ret;
      |         ^~~